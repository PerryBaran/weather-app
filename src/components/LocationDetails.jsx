import React from "react";
import PropTypes from "prop-types";

function LocationDetails({ location, errMessage }) {
  const { city, country } = location;

  return <h1>{errMessage || `${city}, ${country}`}</h1>;
}

LocationDetails.defaultProps = {
  errMessage: "",
};

LocationDetails.propTypes = {
  errMessage: PropTypes.string,
  location: PropTypes.shape({
    city: PropTypes.string,
    country: PropTypes.string,
  }).isRequired,
};

export default LocationDetails;
