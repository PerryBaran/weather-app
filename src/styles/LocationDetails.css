h1 {
  font-size: 3rem;
  padding-left: 5px;
}