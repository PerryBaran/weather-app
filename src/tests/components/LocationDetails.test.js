import React from "react";
import { render } from "@testing-library/react";
import LocationDetails from "../../components/LocationDetails";

describe("LocationDetails", () => {
  it("renders the correct city and location props", () => {
    const location = {
      city: "Leeds",
      country: "UK",
    };
    const { getByText } = render(<LocationDetails location={location} />);

    expect(getByText("Leeds, UK")).toBeInstanceOf(HTMLHeadingElement);
  });
});
