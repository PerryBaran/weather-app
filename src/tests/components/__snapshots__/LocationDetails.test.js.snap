// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocationDetails snapshot 1`] = `
<DocumentFragment>
  <h1>
    Leeds, GB
  </h1>
</DocumentFragment>
`;
